tests/conftest.py

```diff
@@ -88,19 +88,20 @@ def func(value: Union[str, List[Any]]) -> str:
             alias = f" AS {value[2]}" if len(value) > 2 else ""
             return f"{value[0]}({children}){alias}"
 
-        def literal(value: Union[str, List[Any]]) -> str:
-            if isinstance(value, list):
+        def literal(value: Union[str, Sequence[Any]]) -> str:
+            if isinstance(value, (list, tuple)):
                 return f"tuple({','.join(list(map(literal, value)))})"
 
-            try:
-                float(value)
+            if isinstance(value, (int, float)):
                 return f"{value}"
-            except ValueError:
+            else:
                 escaped = value.replace("'", "\\'")
                 return f"'{escaped}'"
 
         sample = legacy.get("sample")
-        sample_clause = f"SAMPLE {sample}" if sample else ""
+        sample_clause = ""
+        if sample is not None:
+            sample_clause = f"SAMPLE {float(sample)}" if sample else ""
```

Comment:
This is because the Entity data source expects the sample to be a float type. We can change that but I didn't want to change more than I needed too in this PR.

Reply:
I think the type is fine, Python considers int to be an acceptable type of float